tensorflow/core/kernels/data/fixed_length_record_dataset_op.cc

```diff
@@ -0,0 +1,512 @@
+#include "tensorflow/core/lib/io/buffered_inputstream.h"
+#include "tensorflow/core/lib/io/inputbuffer.h"
+#include "tensorflow/core/lib/io/random_inputstream.h"
+#include "tensorflow/core/lib/io/zlib_compression_options.h"
+#include "tensorflow/core/lib/io/zlib_inputstream.h"
+
+namespace tensorflow {
+namespace data {
+
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kDatasetType;
+/* static */ constexpr const char* const FixedLengthRecordDatasetOp::kFileNames;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kHeaderBytes;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kRecordBytes;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kFooterBytes;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kBufferSize;
+/* static */ constexpr const char* const
+    FixedLengthRecordDatasetOp::kCompressionType;
+
+constexpr char kFixedLengthRecordDataset[] = "FixedLengthRecordDataset";
+constexpr char kCurrentFileIndex[] = "current_file_index";
+constexpr char kCurrentPos[] = "current_pos";
+constexpr char kZLIB[] = "ZLIB";
+constexpr char kGZIP[] = "GZIP";
+
+class FixedLengthRecordDatasetOp::Dataset : public DatasetBase {
+ public:
+  explicit Dataset(OpKernelContext* ctx, std::vector<string> filenames,
+                   int64 header_bytes, int64 record_bytes, int64 footer_bytes,
+                   int64 buffer_size, const string& compression_type,
+                   int op_version)
+      : DatasetBase(DatasetContext(ctx)),
+        filenames_(std::move(filenames)),
+        header_bytes_(header_bytes),
+        record_bytes_(record_bytes),
+        footer_bytes_(footer_bytes),
+        buffer_size_(buffer_size),
+        compression_type_(compression_type),
+        op_version_(op_version) {}
+
+  std::unique_ptr<IteratorBase> MakeIteratorInternal(
+      const string& prefix) const override {
+    name_utils::IteratorPrefixParams params;
+    params.op_version = op_version_;
+    if (compression_type_.empty()) {
+      return absl::make_unique<UncompressedIterator>(
+          UncompressedIterator::Params{
+              this, name_utils::IteratorPrefix(kDatasetType, prefix, params)});
+    } else {
+      return absl::make_unique<CompressedIterator>(CompressedIterator::Params{
+          this, name_utils::IteratorPrefix(kDatasetType, prefix, params)});
+    }
+  }
+
+  const DataTypeVector& output_dtypes() const override {
+    static DataTypeVector* dtypes = new DataTypeVector({DT_STRING});
+    return *dtypes;
+  }
+
+  const std::vector<PartialTensorShape>& output_shapes() const override {
+    static std::vector<PartialTensorShape>* shapes =
+        new std::vector<PartialTensorShape>({{}});
+    return *shapes;
+  }
+
+  string DebugString() const override {
+    name_utils::DatasetDebugStringParams params;
+    params.op_version = op_version_;
+    return name_utils::DatasetDebugString(kDatasetType, params);
```

Comment:
Note that the debug string will be `FixedLengthRecordDatasetV2Op::Dataset` for `FixedLengthRecordDatasetV2` Op.

Reply:
SGTM